src/sas/sascalc/calculator/ausaxs/ausaxs_sans_debye.py

```diff
@@ -0,0 +1,138 @@
+import ctypes as ct
+import numpy as np
+import logging
+import multiprocessing
+import importlib.resources as resources
+from enum import Enum
+from sas.sascalc.calculator.ausaxs.sasview_sans_debye import sasview_sans_debye
+
+# we need to be able to differentiate between being uninitialized and failing to load
+class lib_state(Enum):
+    UNINITIALIZED = 0
+    FAILED = 1
+    READY = 2
+
+def _attach_hooks():
+    ausaxs = None
+    ausaxs_state = lib_state.UNINITIALIZED
+    from sas.sascalc.calculator.ausaxs.architecture import get_shared_lib_extension
+
+    # as_file extracts the dll if it is in a zip file and probably deletes it afterwards,
+    # so we have to do all operations on the dll inside the with statement
+    with resources.as_file(resources.files("sas.sascalc.calculator.ausaxs.lib")) as loc:
```

Comment:
This could be `os.walk('./ausaxs/')`  or `f for f in pathlib.Path().iterdir('./ausaxs/') if f.is_file()`which would eliminate the need to know the file extension. If multiple files are in the directory, each of them would be tried, so that might slow the calculation down, but nothing else should be there.

Reply:
Will this also work inside a zip file? I specifically chose this approach after a lot of discussions at the camp because it'll be able to handle that.

Reply:
I obviously didn't read your comment close enough. Ignore.